{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘igraph’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:tidyr’:\n",
      "\n",
      "    crossing\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    as_data_frame, groups, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    decompose, spectrum\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    union\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "source(\"~/software/notebook_assist/functions.R\")\n",
    "source(\"~/software/notebook_assist/packages.R\")\n",
    "library(igraph)\n",
    "library(pheatmap)\n",
    "setwd(\"/projects/CARDIPS/analysis/epigenome_resource/\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "theme_bw2 = theme_bw() +\n",
    "    theme(\n",
    "        axis.text = element_text(size = 12),\n",
    "        strip.text = element_text(size = 12),\n",
    "        axis.title = element_text(size = 12),\n",
    "        legend.text = element_text(size = 12),\n",
    "        legend.title = element_text(size = 12),\n",
    "        legend.position = \"top\"\n",
    "    )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load all lead variants from iPSCORE and GTEx QTLs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "leads = fread(\"analyses/jennifer/ipscore_unique_qtls/mashr/2024_0911/input/ipscore_gtex_leadvars_stats.txt\",\n",
    "                     sep=\"\\t\",\n",
    "                     data.table = F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "33793"
      ],
      "text/latex": [
       "33793"
      ],
      "text/markdown": [
       "33793"
      ],
      "text/plain": [
       "[1] 33793"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "250564"
      ],
      "text/latex": [
       "250564"
      ],
      "text/markdown": [
       "250564"
      ],
      "text/plain": [
       "[1] 250564"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "281938"
      ],
      "text/latex": [
       "281938"
      ],
      "text/markdown": [
       "281938"
      ],
      "text/plain": [
       "[1] 281938"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "length(unique(mashr$gene_id))\n",
    "length(unique(mashr$variant_id))\n",
    "\n",
    "nrow(mashr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "mashr = fread(\"analyses/jennifer/ipscore_unique_qtls/mashr/2024_0911/results/mashr_results.txt\",sep=\"\\t\",data.table=F)\n",
    "\n",
    "mashr$gene_id = str_extract(mashr$gene_variant, \"^ENSG[0-9]+\") \n",
    "mashr$variant_id = str_extract(mashr$gene_variant, \"VAR_[0-9]+_[0-9]+_[A-Z]_[A-Z]\") \n",
    "\n",
    "\n",
    "mashr$min_ipscore_lfsr = rowMins(data.matrix(mashr[,6:8]))\n",
    "mashr$min_adult_lfsr    = rowMins(data.matrix(mashr[,c(9:55)]))\n",
    "mashr$min_lfsr         = rowMins(data.matrix(mashr[,c(6:55)]))\n",
    "\n",
    "mashr2 = mashr[ mashr$min_lfsr < 0.05 & mashr$ngtex_tested > 0 & mashr$nipscore_tested > 0 , ]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 402,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "281938"
      ],
      "text/latex": [
       "281938"
      ],
      "text/markdown": [
       "281938"
      ],
      "text/plain": [
       "[1] 281938"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "102375"
      ],
      "text/latex": [
       "102375"
      ],
      "text/markdown": [
       "102375"
      ],
      "text/plain": [
       "[1] 102375"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(mashr) \n",
    "nrow(mashr2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       " Adult   EDev Shared \n",
       " 27881   2299  72195 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "mashr2$Category = ifelse(mashr2$min_adult_lfsr < 0.05 & mashr2$min_ipscore_lfsr < 0.05, \"Shared\",\n",
    "                              ifelse(mashr2$min_adult_lfsr > 0.05 & mashr2$min_ipscore_lfsr < 0.05, \"EDev\",\n",
    "                                    ifelse(mashr2$min_adult_lfsr < 0.05 & mashr2$min_ipscore_lfsr > 0.05,\n",
    "                                           \"Adult\", \"Not Significant\")))\n",
    "\n",
    "mashr2$variant_id = gsub(\"VAR_\",\"\",str_extract(mashr2$gene_variant, \"VAR_.*\"))\n",
    "table(mashr2$Category)\n",
    "\n",
    "kgenomes = fread(\"/projects/CARDIPS/analysis/epigenome_resource/analyses/jennifer/gwas_independent/reference/combined.renamed.bim\",\n",
    "                 sep=\"\\t\",\n",
    "                data.table=F)\n",
    "\n",
    "mashr2$in_kg = ifelse(mashr2$variant_id %in% kgenomes$V2, \"TRUE\",\"FALSE\")\n",
    "mashr3 = mashr2[ mashr2$in_kg == \"TRUE\",]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "write_snps = lapply(unique(mashr3$gene_id), function(gene){\n",
    "    genedf = mashr3[ mashr3$gene_id == gene,]\n",
    "    snpid = gsub(\"VAR_\",\"\",genedf$variant_id)\n",
    "    if(length(unique(snpid)) > 1) {\n",
    "        writeLines(unique(snpid),\n",
    "               paste0(\"analyses/tim/ld_modules/mashr_ld/snps/\",gene,\"_snps.txt\"))\n",
    "        }\n",
    "})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run 03.8.calculate_egene_ld_mashr.sh to identify SNP-eGene pairs representing the same signal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "10343"
      ],
      "text/latex": [
       "10343"
      ],
      "text/markdown": [
       "10343"
      ],
      "text/plain": [
       "[1] 10343"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ld.fs = list.files(\"analyses/tim/ld_modules/mashr_ld/ld\", full.names = T, pattern =\".ld\")\n",
    "length(ld.fs)\n",
    "\n",
    "psize(10,10)\n",
    "read_ld = rbindlist(lapply(ld.fs, function(ld.f){\n",
    "    gene = gsub(\".ld\",\"\",gsub(\"analyses/tim/ld_modules/mashr_ld/ld/\",\"\",ld.f))\n",
    "    all.ld = fread(ld.f,sep=\" \",data.table=F)[,c(\"SNP_A\",\"SNP_B\",\"R2\")]\n",
    "    all.ld$dist = abs(as.numeric(gsub(\"[0-9]+_\",\"\",str_extract(all.ld$SNP_A, \"[0-9]+_[0-9]+\")))-\n",
    "                     as.numeric(gsub(\"[0-9]+_\",\"\",str_extract(all.ld$SNP_B, \"[0-9]+_[0-9]+\"))))\n",
    "\n",
    "    if (nrow(all.ld) > 0 ){\n",
    "    in.ld = all.ld[ all.ld$R2 >= 0.8 & all.ld$dist < 100000,]\n",
    "    verts = data.frame(ids = unique(c(all.ld$SNP_A,all.ld$SNP_B)))\n",
    "    g = graph_from_data_frame(in.ld, directed = FALSE, vertices = verts)\n",
    "    cl = cluster_louvain(g)\n",
    "    verts$gene_id = gene\n",
    "    verts$cluster_id = paste(gene,cl$membership,sep=\"_\")\n",
    "    verts$variant_id = paste(\"VAR\",verts$ids,sep=\"_\")\n",
    "    return(verts)}\n",
    "}))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Add back SNPs that are the only lead variant for a given eGene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "single_qtl = mashr3[ mashr3$gene_id %in% names(table(mashr3$gene_id))[ table(mashr3$gene_id) == 1],]\n",
    "single_qtl$ids = gsub(\"VAR_\",\"\",single_qtl$variant_id)\n",
    "single_qtl$cluster_id = paste(single_qtl$gene_id,1,sep=\"_\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "630"
      ],
      "text/latex": [
       "630"
      ],
      "text/markdown": [
       "630"
      ],
      "text/plain": [
       "[1] 630"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "mashr3$variant_id = str_extract(mashr3$gene_variant, \"VAR_.*\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "mashr4 = rbind(merge(mashr3,read_ld,by=c(\"gene_id\",\"variant_id\")),\n",
    "                       single_qtl)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Identify QTL clusters that have more than one Category and mark as Shared"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "cats = unique(mashr4[ ,c(\"Category\",\"cluster_id\")])\n",
    "cats_dups = cats[ cats$cluster_id %in% names(table(cats$cluster_id))[ table(cats$cluster_id) > 1],]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "mashr4$AdjCategory = ifelse(mashr4$cluster_id %in% cats_dups$cluster_id, \"Shared\",mashr4$Category)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "mashr5 = unique(mashr4[,c(\"gene_id\",\"ntissues_tested\",\"ngtex_tested\",\n",
    "                                          \"min_lfsr\",\"min_adult_lfsr\",\"min_ipscore_lfsr\",\"gene_variant\",\"cluster_id\",\"AdjCategory\")])\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Identify most significant SNPs in each QTL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "mashr6 = mashr5 %>% group_by(cluster_id) %>% top_n(1,-log10(min_lfsr)) %>% top_n(1,ntissues_tested)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 403,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "67269"
      ],
      "text/latex": [
       "67269"
      ],
      "text/markdown": [
       "67269"
      ],
      "text/plain": [
       "[1] 67269"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(mashr6)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select random SNP-eGene pair when there are equally significant pairs in a QTL signal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "mashr7 = mashr6 %>% group_by(cluster_id) %>% sample_n(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "57364"
      ],
      "text/latex": [
       "57364"
      ],
      "text/markdown": [
       "57364"
      ],
      "text/plain": [
       "[1] 57364"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(mashr7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "10971"
      ],
      "text/latex": [
       "10971"
      ],
      "text/markdown": [
       "10971"
      ],
      "text/plain": [
       "[1] 10971"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "57364"
      ],
      "text/latex": [
       "57364"
      ],
      "text/markdown": [
       "57364"
      ],
      "text/plain": [
       "[1] 57364"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       " Adult   EDev Shared \n",
       " 20539   2207  34618 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(mashr7$gene_id))\n",
    "nrow(mashr7)\n",
    "table(mashr7$AdjCategory)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "lead_stats_ipsc_var = leads[leads$gene_variant %in% mashr7$gene_variant,]\n",
    "lead_stats_cast = reshape2::dcast(gene_variant ~ tissue, data = lead_stats_ipsc_var, fun.aggregate = mean, value.var=\"beta\")\n",
    "rownames(lead_stats_cast) = lead_stats_cast$gene_variant\n",
    "lead_stats_cast$gene_variant = NULL\n",
    "lead_stats_mat = data.matrix(lead_stats_cast)\n",
    "\n",
    "# Calculate row means considering NA values\n",
    "row_means <- rowMeans(lead_stats_mat, na.rm = TRUE)\n",
    "\n",
    "# Replace NA with row means\n",
    "lead_stats_mat = t(apply(lead_stats_mat, 1, function(x) {\n",
    "  x[is.na(x)] = row_means[which(is.na(x))] \n",
    "  return(x)\n",
    "}))\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "7165"
      ],
      "text/latex": [
       "7165"
      ],
      "text/markdown": [
       "7165"
      ],
      "text/plain": [
       "[1] 7165"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1919"
      ],
      "text/latex": [
       "1919"
      ],
      "text/markdown": [
       "1919"
      ],
      "text/plain": [
       "[1] 1919"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "9229"
      ],
      "text/latex": [
       "9229"
      ],
      "text/markdown": [
       "9229"
      ],
      "text/plain": [
       "[1] 9229"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(mashr7$gene_id[ mashr7$AdjCategory == \"Adult\"]))\n",
    "length(unique(mashr7$gene_id[ mashr7$AdjCategory == \"EDev\"]))\n",
    "length(unique(mashr7$gene_id[ mashr7$AdjCategory == \"Shared\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "figure3a = as.data.frame(table(mashr7$AdjCategory))\n",
    "figure3a$Specificity = factor(figure3a$Var1, levels = c(\"EDev\",\"Adult\",\"Shared\"))\n",
    "figure3a$No_eQTLs = figure3a$Freq\n",
    "fwrite(figure3a[,3:4], \"~/projects/Resource/notebooks/toGitHub/sourcedata/SOURCEDATA.FIGURE3A.txt\",\n",
    "      sep=\"\\t\",row.names=F,quote=F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAADAFBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OU\nlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWm\npqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4\nuLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnK\nysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc\n3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u\n7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////i\nsF19AAAACXBIWXMAABJ0AAASdAHeZh94AAAfGElEQVR4nO2dCXxTRR7HpxQREAGxsiArioKr\niMcKq+DFsR676rSFUkoBkVIKBYqi0FJRQWo5lkMQWMADRAQRFVFEUAHdXRBEROVQECgU5TRt\noS096JHZN5OXNglt36Tv5T9p+v9+Pib/NyTzy/y/5uUlTV8JQwIaovoBIL4FBQc4KDjAQcEB\nDgoOcFBwgIOCAxwUHOCACB4fagGPUytm8QpaYzPjYAUP/c38HCW2HPOTeEmhLR88M89WZH6S\notCyEgVXAQqWBAXLg4IBQcGSoGB5UDAgKFgSFCwPCgYEBUuCguVBwYCgYElQsDwoGBAULAkK\nlgcFA4KCJUHB8qBgQFCwJChYHhQMCAqWBAXLg4IBQcGSoGB5UDAgKFgSFCwPCgYEBUuCguVB\nwYCgYElQsDwoGBAULAkKvpgdD7do2mVV2ebg+8SVfVGH+jdOcSqevtBxfTi6dePOK72YHAXL\n4UPBW+tcFZvUjszRN98hbcX1EHLPqG5kqGPwXLMEcX2kSaMhSbeRVPnZUbAcPhR8X9ND2u74\n+sZ2sZXe2CF4Oxmp7aKjiPZvLHPDvcQhOJbs1NZ/9yV50rOjYDl8J7i0waP8KoEcEVv3/e0W\nIbh//bOa4MNzfmEsg2g4BN8Twi8nkB+lp0fBcvhOcMGb3/CriKA/+FVqw1/vEIKbPlJ2kFW8\ndesnuuCEoAPa5YP1C6SnR8Fy+PYo+vS2ycH9eLHzkgVMCC4gQ1bd06jDJP0g67Qu+FSnFhNm\nP9hgVWUTXQwKlsO3gu8gpCt/Up6/8R/MITidtAkOH3UP+YfjBk7B9il8d93lhPzUKFgO3wre\nuHhw3a6FjMVdeVIXvJ+QtdouOpasETdwCk4kTx0990HzNhnSU6NgOXz+Qcdk8hZbTz5guuAM\n0oW/Bh8iT4t/1QWfCH6EX60hKdLzQgseciDXNDm2LPOTeMk521kfzbx7kWjJdjIqdwLRuT43\np25k7llbdiYJFzdKI8P41ScklV+dIWHS02dRWMFxR4pMU2g7Z34SL8mz5fpo5nXkVX61mUwr\n+upFTotmL84uKup0e1GuLf9HMl7c6Hcygl/tIWP51WEySnr6PGDBuIv2JLvhrdrxVWkvsl0f\ncLxNWkJW5NkKIoK+FWP6LrqkVchx7UhrCFkrPT2+Bsvhw9fgGaTts+PuJMOd2w7BBbcE9Rp9\nJxntGHMeZH1R94rhyXeRvvKzo2A5fHmQtfzuRs3uXV626RDM/ki45fLOS/Uxp2C2J/yaxne/\nZpefHAXLYVpwNBCeuShYDhQsCQpGwYagYBRsOShYEhSMgg1BwSjYclCwJCgYBRuCglGw5aBg\nSVAwCjYEBaNgy0HBkqBgFGwICkbBloOCJUHBKNgQFIyCLQcFS4KCUbAhKBgFWw4KlgQFo2BD\nUDAKthwULAkKRsGGoGAUbDkoWBIUjIINQcEo2HJQsCQoGAUbgoJRsOWgYElQMAo2BAWjYMtB\nwZKgYBRsCAr2reDTM4b2TnjL8VcD1idFJa1ncqUABfu94JOR4c8vGENHXtDqBTR+djxdxGRK\nByjY7wVPpTu0yzfpOsbS6MQSVjIhNF2i1EHBfi/4iTh+mU7nMDaTHtXKw3S2RKmDgv1dsH0d\nfwKzn/h+NzZGDA2Mkyh1ULC/C+YUpG8bFpnO7D2TxOaYCONSY3uoRvTeLPPYMiyYxDsybZnm\nJoAS7Jl75vFqCF5AacRPjJ2jL4vNSTTHsNQut3TXiNqTYR6bzYJJgCOhBHvmnqqO4CNbVsX0\n+pZl0VTdX5Zh6bwr7qJ9jGdudT/oyOwziNnDkkU9NtxuWDrvh4L9XfCpDY73PONprvPgKSaG\nGZc6KNjfBe+nr4vrhEg7m0GPa9UxOpMZlzoo2N8FlwyIOqVd/Y+/vO6ls7T3TdPpzxKlDgr2\nd8FsS2jvaQvH0yf4X1dLoc8vS6aTmUzpAAX7vWC2Z2K/3k8tzhX3XpkYlbiqWKp0gIL9X7Ap\nUDAKNgQFo2DLQcGSoGAUbAgKRsGWg4IlQcEo2BAUjIItBwVLgoJRsCEoGAVbDgqWBAWjYENQ\nMAq2HBQsCQpGwYagYBRsOShYEhSMgg1BwSjYclCwJCgYBRuCglGw5aBgSVAwCjYEBaNgy0HB\nkqBgFGwICkbBloOCJUHBKNgQFIyCLQcFS4KCUbAhKBgFWw4KlgQFo2BDUDAKthwULAkKRsGG\noGA/Fhx3tMQ0F2zZ5ifxknzbeXMTQAn2zC0AFhz78znTnLVlmJ/ES7JsWeYmgBLsmWurzhnf\nTYC7aB/jmYuvwXKgYElQMAo2BAWjYMtBwZKgYBRsCApGwZaDgiVBwSjYEBSMgi0HBUuCglGw\nISgYBVsOCpYEBaNgQ1CwdYLtxy6YW5cnKNhvBP9n8FF2vAO59EU7sxAU7C+CN9Qh+1g0eagT\nWWluZe6gYH8RfH+T7SyvQTd2oe395lbmDgr2F8HNnmRsE1nMWHxzcytzBwX7i+Am/Rh7jhxk\nbMhl5lbmDgr2F8F/u/pC0TXXazdo297cytxBwf4ieCn5y3UkhX15O5lkbmXuoGB/EcwmhwT3\nzGcvkLBccytzBwX7jWDt37T/DqWZW5cnKNiPBDtY2NfcytxBwX4neJCln1GjYBRsCApGwZaD\ngiXxd8EHY2+7rENKAS8LxrYPeWyXCLyECEq0esfDLZp2WSVum9732j+FfS45MQqWx4eCDzat\nG5rcmTxk11I61wkb0bLhFm30CPlbn56RffuWMra1zlWxSe3IHG10d+PGw0b/OWiN3MwoWB4f\nCh5IVjBmH0zeZ+wjkspYWrOHtdGN5FPnLvq+pocYy7u+sfZ/QJeGBxjLbHeL3Mx+JXimK51q\nk+DmrfnlYTKKsR4N+I46gWgWF5EDuuDSBo8yMXqE/UGG8nIx2So1s18JJu5Uv18X49+Ci28e\nzq/SyCDGmvDnLvuEP6UTgzNWTPuc+y548xs+GhH0B9tCpvPyW/Kq1NR+JXiRO9Vv2MX4t2Cd\naWQRyyeDefk9mcVYr3oh2v/n7b5z/OvpbZOD+/H/CxL41ioyXmpOvxLsQ2qC4CXB7fK13fQY\nXh8lYxm7nYw78Ov8y652xN5BSFft2WxvfcUJxs7/lQyRmtSvBLcdY65DVeD/gn+LJDelMXaI\nm+UH0NrL8cat/H3wK/rOeOPiwXW7FjK2ts6fnnqmTRt+Awn8SjAZYK5HVeD3ghdfXi+RH0/l\nkFi+uYtMEcOa4N8dI5zJ5C3tcmuPK24Y/i2ZLDUvCpbHl4KHkXsPOaqG4nj5M6FSCC4k/dih\npb/zrd3kWecdNvCvNUmAguXxoeC5JK5ILx+6nFfPkv3sh05LuOAftefq52Q+/7f/khnaS/Va\nXg4PlluRfwnuneFKtdtVAf4tuLR16xJnvZIs1I6YW97PWH6T6zILbdm03gGW3fBW7fiqtBfZ\nztgjl/3GP87qKTe1fwmure+D95EbBwne1nbKd1yaMOXm4M3a8Buk1eiR7clUrZxB2j477k7C\n3y7vqNcyafAVzQ7ITe1fgkM6u+Jxl5w34iLiZpwS9fqkqKT1TK4U+Lfgdc7/p+O1jYyR7UNC\nN4vxDT2vadb9M1Euv7tRs3uXi3Jr18YhA9Ilp/YvwVW9BucMpSPmTQiN4N/lWUDjZ8dT8UmI\nYenALwRDNTtaQabnWr0X/LbwtTP0KcbS6MQSVjIhNF2i1EHBcJEC7wUPDxUfu79Az7GZ9KhW\nHaazmXGpg4LhIgXeCx7xgriaRI+z2BhRDoxjxqUOCoaLFFQoOH6JYY/ORTxRYu+ZJOoxEcyw\n1DizUSPmcKFp8m1nzU0A1exoBZmea82lFQg25vhQuomdoy+LjUk0x7DULr/qqNFnt009UM2O\nVpDpudaTlZzxvWTNsO5/7R7/SSljb2Z76i1YERGxjrEsmqr7yzIstcsTqzWePFRgmjzbWXMT\nQDU7WkGm51pzKn4Gf9NWf0N483dLxBt8V74fTFOOa9f2sGSxPTbcblg674uvwXCRgop/Hvx5\nXdJt5U+ZP67sTurVe7TU/S4raNw+R6UfPMXESJQ6KBguUlCh4Mymwc7zNrxKyFH3e2ymqXl6\nOYPyJ/IxOlOi1EHBcJGCCgW/RFL0qviBuvwHJy7Y4yPPO+u9dJY2MJ3+LFHqoGC4SEGFgjs2\nKNarb+p/ctVdbnc4Q/tPdKAde6XQ55clU/ETb8PSAQqGixRUKLhJ+c8XTjHqfo6O3dRJhnbv\nlYlRiavE/w2GpQMUDBcpqFDwpY+53CKsvrluuoOC4SIFFQq+tpXLLa5pY66b7qBguEhBhYKf\nIN+V1bv4l8CtAwXDRQoqFLyF3OBs4fkbyTfmuukOCoaLFFT8Qcdgcv2Hovi4LYnzvIspUDBc\npKBiwUUDCQm5r9/9VxEysMjzLqZAwXCRgsp+dWVdD/5Lz5d0X2eulReBguEiBZX/blLevq17\nzzOrQcFwkYKqfvksb892c42sABQMFymoXHB6L20nzSb0P2GumR6gYLhIQaWCT15D7ulO2Nyg\nVifNddMdFAwXKahU8EiyjC3XhjYEjzDXTXdQMFykoFLB13ZnQjALbWeum+6gYLhIQaWCLxum\nCx4eeCcEh2q2Xwu++y5d8J0dzXXTHRQMFymoVHAqebmUC04lyea66Q4KhosUVCq45AHStgsZ\n0ZHcWmCum+6gYLhIQeXvgy/MaU0IufIFa39VEwXDRQqqPI1S7s+Z5jp5MSgYLlJQK8+TBdVs\nFFw9ULBcpAAFA3VbQaQABQN1W0GkAAUDdVtBpAAFA3VbQaQABQN1W0GkAAUDdVtBpMBQ8Nk7\n7jDXTXdQMFykwFBwRgCeyhCq2TVCcNGmTea66Q4KhosU4GswULcVRAqqFJz3q+Vn/UTBcJGC\nygXnplxNCGmZYu2X31EwXKSgUsH5HUiLXiN7tyK3F5rrpjsoGC5SUKngRJLMzV54njxnrpvu\noGC4SEGlgv/q/K7dXfilOyu6rSBSUPnXZofqxfBG5rrpDgqGixRUKvjW+/Wi623muukOCoaL\nFFTxqytzxAkm5zv+RJ9VoGC4SEGlgrOvJ7cmpI66jbQ5Z66b7qBguEhB5e+DTw0Xv+M/zNJf\nLkTBgJGCqj7JKjr0n4PWnqEDBUNGCqA/i47dn2OabFumuQmgmh2tINNzrZkVnfE9xB0rBccd\nLTbNBVu2uQmgmh2tINNzrfkVPYNvcqEe/jy4+ijI9FyrwS76RG9y6UvmuukOCoaLFFQpuHRu\nY/L3g+aa6QEKhosUVCX4+46k+TvmWnkRKBguUlC54Jyng4Pissx18mJQMFykoFLBq1uRDpae\nZ9YBCoaLFFQiOP0x0nBaseeNLQAFw0UKKhRcPL0heeyouSZWAgqGixRUKPhWEjRs46YyzHXT\nHRQMFymo+M/LumOum+6gYLhIQYWCU90x1013UDBcpAC/+A7UbQWRAhQM1G0FkQIUDNRtBZEC\nFAzUbQWRAhQM1G0FkQIUDNRtBZECFAzUbQWRAhQM1G0FkQIUDNRtBZECFAzUbQWRAhQM1G0F\nkQIUDNRtBZECFAzUbQWRAhQM1G0FkQIUDNRtBZECFAzUbQWRAhQM1G0FkQIUDNRtBZECFAzU\nbQWRAhQM1G0FkQIUDNRtBZECFAzUbQWRAhQM1G0FkQIUDNRtBZECFAzUbQWRAhQM1G0FkQIU\nDNRtBZECFAzUbQWRAhQM1G0FkQIUDNRtBZECFAzUbQWRguoJPrmuur1FwXCRguoJfr2vXqxP\nikpaL1kKUDBcpKBagn/spQteQONnx9NFUqUDFAwXKaiG4FnxlDoEp9GJJaxkQmi6RKmDguEi\nBdUQPCUlpY9D8Ex6VLs8TGdLlDooGC5SUL3X4ASH4NgYcTUwTqLUQcFwkQIzgu09k8TWmAjj\nUmN7qEb03izz2DLM3R+q2dEKMj3XeqaiM75LCj5HXxZbk2iOYaldbumuEbUnwzw2m7n7QzU7\nWkGm51pPmRCcRVN1f1mGpfOeuIuGixSY2kWHJYutseF2w9J5TxQMFykwdZClHzzFxEiUOigY\nLlJgSvAMely7PEZnSpQ6KBguUmBK8F46S9tRT6c/S5Q6KBguUmBKMEuhzy9LppOlSgcoGC5S\nYE5w0crEqMRVxVKlAxQMFynAnwcDdVtBpAAFA3VbQaQABQN1W0GkAAUDdVtBpAAFA3VbQaQA\nBQN1W0GkAAUDdVtBpAAFA3VbQaQABQN1W0GkAAUDdVtBpAAFA3VbQaQABQN1W0GkAAUDdVtB\npAAFA3VbQaQABQN1W0GkAAUDdVtBpAAFA3VbQaQABQN1W0GkAAUDdVtBpAAFA3VbQaQABQN1\nW0GkAAUDdVtBpAAFA3VbQaQABQN1W0GkAAUDdVtBpAAFA3VbQaQABQN1W0GkAAUDdVtBpAAF\nA3VbQaQABQN1W0GkAAUDdVtBpAAFA3VbQaQABQN1W0GkAAUDdVtBpAAFA3VbQaQABQN1W0Gk\nAAUDdVtBpAAFA3VbQaQABQN1W0GkAAUDdVtBpAAFA3VbQaQAWnDc0WLTXLBlm5sAqtnRCjI9\n15oPLDh2f45psm2Z5iaAana0gkzPtWZW64zv1Qd30XCRAnwNBuq2gkgBCgbqtoJIAQoG6raC\nSAEKBuq2gkgBCgbqtoJIAQoG6raCSAEKBuq2gkgBCgbqtoJIAQoG6raCSAEKBuq2gkgBCgbq\ntoJIAQoG6raCSIH/Cz4Ye9tlHVIKXEouuGBs+5DHdmljh4lOW/kpoZqNgiU42LRuaHJn8pC9\nvCy25ZR0rhM2omXDLYyd7ivoE9xdfk6oZqNgCQaSFYzZB5P3y8v3bDkfkVTG0po9XHazRfX2\nyc8J1WwULEHz1vzyMBlVXibYcno04PvsBHJAv9WpppO8mBOq2SjYmOKbh/OrNDKovHzSltNE\nPHc/4U9pQdTNRV5MCtVsFCzNNLKovFxgO0MG8/J7Mssxto186M1sUM1GwbIsCW6XX17m2naT\nMbw+SsY6BnvcbvdmOqhmo2A5foskN6W5lCW2nxxmj/BXZo2vyGqvJoRqNgqWYvHl9RLzXcsS\n20kSyzd3kSliOOyq4krvXRFQzUbBMgwj9x5yL0tsOQ0f5dufkbf41ZlLRnk3JVSzUbAEc0lc\nkUepCX7ocl49S/bz7VfIt97NCdVsFGxMaevWJZ6lJnglWcjY6Zb3i+0ejUu9mxSq2SjYmH3k\nxkGCt8vLt2w5hXdcmjDl5uDN/CYF9R82msUDqGajYGPWOX+WEF9eDrPlsIyR7UNChV+2mUz0\n8kFANRsFcxSsHCoSBXMUrBwqEgVzFKwcKhIFcxSsHCoSBXMUrBwqEgVzFKwcKhIFcxSsHCoS\nBXMUrBwqEgVzFKwcKhIFcxSsHCoSBXMUrBwqEgVzFKwcKhIFcxSsHCoSBXMUrBwqEgVzFKwc\nKhIFcxSsHCqyNghenxSVtN5lGwXDRQp8LHgBjZ8dTxeVD6BguEiBbwWn0YklrGRCaHrZCAqG\nixT4VvBMelS7PExnl42gYLhIgW8Fx8aIq4FxZSMoGC5S4FPB9p5J4npMBL88sVrjyUMFHkCt\nXEGkikzP9uZQHwo+R18W15MoP2vKVx01+uy2IZCc9OUZ37Noqi44i1X6DPaePNtZ85N4Sa4t\nBzwz23be/CQ+fQbbw5LF9djwsl/f9YvTKFWDQlu+8Y0sJs/mzWkLKsG3B1n60VVMTNkICpan\nBgieQY9rl8fozLIRFCxPDRC8l87SdtTT6c9lIyhYnhogmKXQ55cl08nlAyhYnpoguGhlYlTi\nKpcTLKBgeWqC4ItAwfKgYEBQsCQoWB4UDAgKlgQFy4OCAUHBkqBgeVAwIChYEhQsDwoGBAVL\nMrTPANP07xttfhKvM/uBZ/br29/8JP2BBedlm2d/t5csmMU71ndbCp75727/s2CWXFjBVnCs\n4wTwzI0d3wHPXNjRy7PoGoCCqwAFA4KCq0eNEWwbtwo8c8+4/4FnfjHuoKXz1RjBSPVAwQEO\nCg5w/FHwIeokhrFfRBEeO/+U74Nn0VCb6/a4GI9/tuBDJs7+KfERA8dv5L8YMI8WeHNPj0ck\ngX8Kjp3l4DUueIhWvNiPRuzyde6FSErXuA5UIPjH2O2mc96nESn/ntyPvlRaawVPL9/4hTr+\nVOGasL55Ps7dQl+iY1wHKhD8Lf3abMzh0BH8t7YKptLVKLhcMHuPrvRx7mR6IIG6vhL4RvBq\nfYrzoSOrEmw/X8FgIAvO6T1Auyz9IDEy9rUsfh6BX/joFrrUqti8XrHsXfqBqE9Mixk07Qxv\nZ0ok3y7ij0IT/AI/IDD5Y6036FeOYt06Ljh/5ejIkV/y7T9eGRERM+2IVs0ZaH89coPLal0f\nkXfUHMEsmZ5nxc/RZ+aPp4P/0J5KS/jgNHrMqthN9G2WTp/i5f6o0PGvxAyMvUjwD6/T+ZtN\nHmr9lw74tOxpO4+mDnlzURTdxv8GZ6+pC8aE9cvkgt8ZMGO/y2pdH5F3+KfgIXMcbHYVPJOm\nsbX0Pa3aTKeyoqihWlXY+xnLYifQdMaG0d+18tlQreH54+hFgq3YRdunUhr58lrHz8jn0VHa\nocU+Pv1rdKc28Bl/fs8JS+C7ifLVuj4i7/BPwU4WuAp+Q2vA4KHil44TexZq7daeu1vpWqtS\nz4WN0C6X0nf5A/gXHznoG8HM/tO8Idrihqyxc8Hf8JGeEzXLX/O1fc9XNIdu4TcsX63rI/IO\n/xRc4S56Fj1cQBO/5ozXnmw76CrGpoefsyp1HZ9Pa6Gm+Wu6UQz1941gzukvkyifcx49wzcj\nJ4rRrB/ejXMIPqltuazW9RF5Rw0SrL0G/1b23N6v7aOf0d65pliWmuSc+ihbQx3vuZ/2ieBS\n/cwHvw6gh5xH0Vzw2ZkDaORz/3II5t8Vclmt6yPyjpoj+HzkAJbjev68V6htG91qVagtdOAb\nnHF0mXZovkmMlR9k5Vgn2N7T+V77bbrZVfDY0EX77OxXh+ACEVq2WtdH5B01R/D7/H1wP8cx\n1acrtIvv6LoZUResCl2tHUNzfqVxLM3xAE6HcsG9+BNut4XP4JER+uehs9yewTZ+LMXYj+WC\nXVbr+oi8o8YIXhfOP8l6h76v1T+E8tNDFEcl9ZlvWejT4vx8GrFa28eG7mDswiR+SDOb7tPK\nceWCN5pNWk9H8yN1tj18aKmL4Bz6glbkJNOPywS7rNblEXmHfwp2vk2ac1wTHKddTxpII77X\n/ik/gT6zMDVsUAa/3Wzq+LDDCo7zgyvBW3Qxf9f54ty4SP6xwnYa9cZb8RF9dME/0GfeKzQX\nZZ9Dez41NXU4jTrGXHfRE+mk9+b3f4GO2ukU7LJal0fkHf4puIy9zp8mxcx1fIZ4Yeno3nEL\nxGc7bCeNq2oar3hXvOV0pMfY2fGpMQOnpC3g7dyU0Iv22zlUF3whJaJfblUTybBj8pM9nxiz\nPJu5Cc5ZEBOVvJktjJrnFOy6WpdH5BX+KNjfsP9RYnwjfwUFBzgoOMBBwQEOCg5wUHCAg4ID\nHBQc4KDgAAcFBzi1UvC2Xn+p36L7YrvxLVk4b9DUFo0KWaN/XDReE6gpj9NKJpP6jw8Lv5L8\nU+ITSC5yJ7lpYkkFgr+8bk1ld/MfaqHgXXVu5l+IyY0g/zK+ccZxxpYR/mWeEzbP8Y/Jct88\nQiuphYKnE8evdZ+tc4u4tht9DXYpWVfhOAr2T0aTZY5i/jzGQoYsaU5a90nj2yVTujS6bpT4\nsWRmfPumPfgXr3sTdh/R0F+DXccf5OMZM8lqMRd5W8liDKmFgt8lzeeW/Ug35PqgNk8+EHTF\nLsYudCWdhnYjrY8xduy64EeGtiVPC5Hrh5DRS0uFYLfxL54iQ5cWppEn+DxdL81WtqAqqYWC\nSyMIaRT6quNvhoSQfxYwtpL8nbE5hP/JtrdJBGNPkI8YK7on6CAXqe+iuWD3cccu+vZmxYyd\nqtNL5ZKqoBYKZvaNQ9poe9c2s0o1wXXEKTEeIwdZ67alvOxSL89W5++8+uy+je6CPcYdgl8i\nX/E99PuqVmNAbRTMOfLmvYT01wRfJzbnkU/Pky7LOd3Jnq0k1Xk7N8Ee4w7Bu/keu2sj+JMe\nylELBZeUOq6/bU6+ZyGdRf0R+ffPxMm2FeRN543dBHuM60fRN1yr7aEHQK7AG2qf4NJ6d+nV\nc9qRb0gbUS4gqzPISOdNNpW/Q3YT7DGuCx5LfppfyRspP6D2CWa3NNBPfjuA7NRegw/zMpzs\nZVd2EqNzJ7DfiDiZ54a6i9wFe4zrgreRlx5oZtHZO6ynFgpeSO7cz6/X1G1boh1FP1bI2IdB\n9zD2POF/rO2LoH6MPRq0gbHiHkEHPI6i3cc/JuJXlO0t29Sx7vu7VlMLBZcOIvVujwi7iTTe\npx1k/blZ29gHg5rsYCynA+k4Iiy41XHG9jcPfnxke/KMxy7aY/wL0ullft6QeEK+Ur2oSqmF\nghlbG351vT/d/SL/cDmk26GwFq0iDvHh/HF3Nrwhnn9OzU4ObNfozjfsnoLdxwser9+Mfyn9\nS9KyVN1iDKiVgl0I6WZ+jl38jZK/goLNz/EssfYEsZaCgs3OkP1DoxuteCQ+orYLbm/6M+QQ\nEvShFY/ER9R2weaZPtx/D6EZCg54UHCAg4IDHBQc4KDgAAcFBzgoOMBBwQEOCg5w/g/qBgSB\n01/APgAAAABJRU5ErkJggg==",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 240,
       "width": 240
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Correlation of EDev, Adult and Shared eQTLs across iPSCORE and GTEx tissues"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The melt generic in data.table has been passed a matrix and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is superseded and is no longer actively developed, and this redirection is now deprecated. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace, i.e. reshape2::melt(edev_cor). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“The melt generic in data.table has been passed a matrix and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is superseded and is no longer actively developed, and this redirection is now deprecated. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace, i.e. reshape2::melt(shar_cor). In the next version, this warning will become an error.”\n",
      "Warning message:\n",
      "“The melt generic in data.table has been passed a matrix and will attempt to redirect to the relevant reshape2 method; please note that reshape2 is superseded and is no longer actively developed, and this redirection is now deprecated. To continue using melt methods from reshape2 while both libraries are attached, e.g. melt.list, you can prepend the namespace, i.e. reshape2::melt(adult_cor). In the next version, this warning will become an error.”\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 3 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>QTL Specificity</th><th scope=col>mean_r</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>EDev  </td><td>0.001188834</td></tr>\n",
       "\t<tr><td>Adult </td><td>0.004061838</td></tr>\n",
       "\t<tr><td>Shared</td><td>0.211740161</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 3 × 2\n",
       "\\begin{tabular}{ll}\n",
       " QTL Specificity & mean\\_r\\\\\n",
       " <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t EDev   & 0.001188834\\\\\n",
       "\t Adult  & 0.004061838\\\\\n",
       "\t Shared & 0.211740161\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 3 × 2\n",
       "\n",
       "| QTL Specificity &lt;fct&gt; | mean_r &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| EDev   | 0.001188834 |\n",
       "| Adult  | 0.004061838 |\n",
       "| Shared | 0.211740161 |\n",
       "\n"
      ],
      "text/plain": [
       "  QTL Specificity mean_r     \n",
       "1 EDev            0.001188834\n",
       "2 Adult           0.004061838\n",
       "3 Shared          0.211740161"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "edev_cor = cor(lead_stats_mat[ rownames(lead_stats_mat) %in% mashr7$gene_variant[ mashr7$AdjCategory == \"EDev\"],])\n",
    "\n",
    "shar_cor = cor(lead_stats_mat[ rownames(lead_stats_mat) %in% mashr7$gene_variant[ mashr7$AdjCategory == \"Shared\"],])\n",
    "adult_cor = cor(lead_stats_mat[ rownames(lead_stats_mat) %in% mashr7$gene_variant[ mashr7$AdjCategory == \"Adult\"],])\n",
    "\n",
    "\n",
    "edev_melt = melt(edev_cor)\n",
    "edev_melt$Type = \"EDev\"\n",
    "shar_melt = melt(shar_cor)\n",
    "shar_melt$Type = \"Shared\"\n",
    "adult_melt = melt(adult_cor)\n",
    "adult_melt$Type = \"Adult\"\n",
    "\n",
    "melted = rbind(rbind(edev_melt,shar_melt),adult_melt) %>% filter(Var1 != Var2)\n",
    "melted$Tissue1 = as.character(melted$Var1)\n",
    "melted$Tissue2 = as.character(melted$Var2)\n",
    "melted$r2 = melted$value**2\n",
    "melted$Var1 = NULL\n",
    "melted$Var2 = NULL\n",
    "melted2 = melted[!duplicated(apply(melted,1,function(x) paste(sort(x),collapse=''))),]\n",
    "melted3 = melted2[ melted2$Tissue1 %in% c(\"iPSC\",\"CVPC\",\"PPC\") |  melted2$Tissue2 %in% c(\"iPSC\",\"CVPC\",\"PPC\"),]\n",
    "melted4 = melted3 [ !(melted3$Tissue1 == \"iPSC\" &  melted3$Tissue2 == \"CVPC\") &\n",
    "                     !(melted3$Tissue1 == \"PPC\" &  melted3$Tissue2 == \"iPSC\") &\n",
    "                     !(melted3$Tissue1 == \"PPC\" &  melted3$Tissue2 == \"CVPC\"),]\n",
    "melted4$`iPSCORE Tissue` = ifelse(melted4$Tissue1 %in% c(\"iPSC\",\"CVPC\",\"PPC\"), melted4$Tissue1,\n",
    "                                  ifelse(melted4$Tissue2 %in% c(\"iPSC\",\"CVPC\",\"PPC\"), melted4$Tissue2, NA))\n",
    "melted4$`iPSCORE Tissue` = factor(melted4$`iPSCORE Tissue`, levels = c(\"iPSC\",\"CVPC\",\"PPC\"))\n",
    "melted4$`QTL Specificity` = factor(melted4$Type, levels = c(\"EDev\",\"Adult\",\"Shared\"))\n",
    "melted4 %>% group_by(`QTL Specificity`) %>% summarise(mean_r = mean(r2))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "fwrite(melted4, \n",
    "      \"~/projects/Resource/notebooks/toGitHub/sourcedata/SOURCEDATA.FIGURE3B.txt\",sep=\"\\t\",row.names=F,quote=F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 3 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>QTL Specificity</th><th scope=col>mean_r</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>EDev  </td><td> 0.02003603</td></tr>\n",
       "\t<tr><td>Adult </td><td>-0.05646204</td></tr>\n",
       "\t<tr><td>Shared</td><td> 0.45889373</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 3 × 2\n",
       "\\begin{tabular}{ll}\n",
       " QTL Specificity & mean\\_r\\\\\n",
       " <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t EDev   &  0.02003603\\\\\n",
       "\t Adult  & -0.05646204\\\\\n",
       "\t Shared &  0.45889373\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 3 × 2\n",
       "\n",
       "| QTL Specificity &lt;fct&gt; | mean_r &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| EDev   |  0.02003603 |\n",
       "| Adult  | -0.05646204 |\n",
       "| Shared |  0.45889373 |\n",
       "\n"
      ],
      "text/plain": [
       "  QTL Specificity mean_r     \n",
       "1 EDev             0.02003603\n",
       "2 Adult           -0.05646204\n",
       "3 Shared           0.45889373"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1.4977869256021e-12"
      ],
      "text/latex": [
       "1.4977869256021e-12"
      ],
      "text/markdown": [
       "1.4977869256021e-12"
      ],
      "text/plain": [
       "[1] 1.497787e-12"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1.12377989562563e-119"
      ],
      "text/latex": [
       "1.12377989562563e-119"
      ],
      "text/markdown": [
       "1.12377989562563e-119"
      ],
      "text/plain": [
       "[1] 1.12378e-119"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7.57729016380492e-121"
      ],
      "text/latex": [
       "7.57729016380492e-121"
      ],
      "text/markdown": [
       "7.57729016380492e-121"
      ],
      "text/plain": [
       "[1] 7.57729e-121"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "melted4 %>% group_by(`QTL Specificity`) %>% summarise(mean_r = mean(value))\n",
    "t.test(melted4$r2[ melted4$`QTL Specificity` == \"EDev\"],\n",
    "      melted4$r2[ melted4$`QTL Specificity` == \"Adult\"])$p.value\n",
    "\n",
    "t.test(melted4$r2[ melted4$`QTL Specificity` == \"EDev\"],\n",
    "      melted4$r2[ melted4$`QTL Specificity` == \"Shared\"])$p.value\n",
    "\n",
    "t.test(melted4$r2[ melted4$`QTL Specificity` == \"Adult\"],\n",
    "      melted4$r2[ melted4$`QTL Specificity` == \"Shared\"])$p.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_info = fread(\"/reference/private/Gencode.v44lift38/gene_info.txt\",sep=\"\\t\",data.table=F)\n",
    "gene_info$simple_gene = gsub(\".[0-9]+$\",\"\",gene_info$gene_id)\n",
    "mashr7$pos = as.numeric(gsub(\"_[A-Z]+\",\"\",str_extract(mashr7$gene_variant, \"[0-9]+_[A-Z]+\")))\n",
    "gene_info_key = gene_info[,c(\"start\",\"end\",\"strand\",\"simple_gene\")] %>% rename(gene_id = simple_gene)\n",
    "gene_info_key$tss = ifelse(gene_info_key$strand == \"+\",gene_info_key$start,gene_info_key$end)\n",
    "mashr8 = merge(mashr7,gene_info_key[,c(\"gene_id\",\"tss\")])\n",
    "mashr8$dist = abs(mashr8$tss - mashr8$pos)\n",
    "\n",
    "mashr9 = aggregate(dist ~   gene_id + AdjCategory, data = mashr8 , FUN = min)\n",
    "psize(3,3)\n",
    "mashr9$Category = factor(mashr9$AdjCategory, levels = c(\"EDev\",\"Adult\",\"Shared\"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "fwrite(mashr9,\"~/projects/Resource/notebooks/toGitHub/sourcedata/SOURCEDATA.FIGURE3C.txt\",sep=\"\\t\",row.names=F,quote=F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1.90508401030251e-212"
      ],
      "text/latex": [
       "1.90508401030251e-212"
      ],
      "text/markdown": [
       "1.90508401030251e-212"
      ],
      "text/plain": [
       "[1] 1.905084e-212"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "wilcox.test(mashr9$dist[ mashr9$AdjCategory == \"EDev\"],\n",
    "           mashr9$dist[ mashr9$AdjCategory == \"Adult\"])$p.value\n",
    "\n",
    "wilcox.test(mashr9$dist[ mashr9$AdjCategory == \"EDev\"],\n",
    "           mashr9$dist[ mashr9$AdjCategory == \"Shared\"])$p.value\n",
    "\n",
    "wilcox.test(mashr9$dist[ mashr9$AdjCategory == \"Adult\"],\n",
    "           mashr9$dist[ mashr9$AdjCategory == \"Shared\"])$p.value\n",
    "\n",
    "wilcox.test(mashr9$dist[ mashr9$AdjCategory == \"Shared\"],\n",
    "           mashr9$dist[ mashr9$AdjCategory != \"Shared\"])$p.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "220.765"
      ],
      "text/latex": [
       "220.765"
      ],
      "text/markdown": [
       "220.765"
      ],
      "text/plain": [
       "[1] 220.765"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "20.125"
      ],
      "text/latex": [
       "20.125"
      ],
      "text/markdown": [
       "20.125"
      ],
      "text/plain": [
       "[1] 20.125"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7.257"
      ],
      "text/latex": [
       "7.257"
      ],
      "text/markdown": [
       "7.257"
      ],
      "text/plain": [
       "[1] 7.257"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "median(mashr9$dist[ mashr9$Category == \"EDev\"])/1000\n",
    "median(mashr9$dist[ mashr9$Category == \"Adult\"])/1000\n",
    "median(mashr9$dist[ mashr9$Category == \"Shared\"])/1000"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate eQTLs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                \n",
       "                 CVPC iPSC  PPC\n",
       "  EDev            900 1068  248\n",
       "  No Association 1116 3042 2076\n",
       "  Shared         2821 4902 3132"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2.25250269479419e-06"
      ],
      "text/latex": [
       "2.25250269479419e-06"
      ],
      "text/markdown": [
       "2.25250269479419e-06"
      ],
      "text/plain": [
       "[1] 2.252503e-06"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "7.90465922105566e-71"
      ],
      "text/latex": [
       "7.90465922105566e-71"
      ],
      "text/markdown": [
       "7.90465922105566e-71"
      ],
      "text/plain": [
       "[1] 7.904659e-71"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1.4497845989729e-15"
      ],
      "text/latex": [
       "1.4497845989729e-15"
      ],
      "text/markdown": [
       "1.4497845989729e-15"
      ],
      "text/plain": [
       "[1] 1.449785e-15"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "filtered_qtls = fread(\"analyses/tim/gwas_coloc/scripts/qtls_090424.txt\",sep=\"\\t\",data.table=F)\n",
    "eqtls = filtered_qtls[ filtered_qtls$type == 0 & grepl(\"^E\",filtered_qtls$element_id),]\n",
    "\n",
    "eqtls$gene_id = gsub(\".[0-9]+$\",\"\",eqtls$element_id)\n",
    "eqtls$variant_id = eqtls$id\n",
    "eqtls$gene_variant = paste(eqtls$gene_id, eqtls$variant_id)\n",
    "\n",
    "\n",
    "eqtls$Category = ifelse(eqtls$gene_variant %in% mashr5$gene_variant[ mashr5$AdjCategory == \"EDev\"], \"EDev\",\n",
    "                       ifelse(eqtls$gene_variant %in% mashr5$gene_variant[ mashr5$AdjCategory == \"Shared\"], \"Shared\",\"No Association\"))\n",
    "\n",
    "table(eqtls$Category, eqtls$tissue)    \n",
    "eqtls$tissue = factor(eqtls$tissue, levels = c(\"iPSC\",\"CVPC\",\"PPC\"))\n",
    "psize(3,3)\n",
    "\n",
    "wilcox.test(abs(eqtls$beta[ eqtls$Category == \"EDev\" & eqtls$tissue == \"iPSC\"]),\n",
    "               abs(eqtls$beta[ eqtls$Category == \"Shared\"& eqtls$tissue == \"iPSC\"]))$p.value\n",
    "wilcox.test(abs(eqtls$beta[ eqtls$Category == \"EDev\" & eqtls$tissue == \"CVPC\"]),\n",
    "               abs(eqtls$beta[ eqtls$Category == \"Shared\"& eqtls$tissue == \"CVPC\"]))$p.value\n",
    "wilcox.test(abs(eqtls$beta[ eqtls$Category == \"EDev\" & eqtls$tissue == \"PPC\"]),\n",
    "               abs(eqtls$beta[ eqtls$Category == \"Shared\"& eqtls$tissue == \"PPC\"]))$p.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "fwrite(eqtls[eqtls$Category %in% c(\"EDev\",\"Shared\"),c(\"element_id\",\"tissue\",\"beta\",\"Category\",\"path\")],\n",
    "       \"~/projects/Resource/notebooks/toGitHub/sourcedata/SOURCEDATA.FIGURE3E.txt\",\n",
    "       sep=\"\\t\",row.names=F,quote=F)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R 4.0.1",
   "language": "R",
   "name": "ir40"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
